sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala

```diff
@@ -255,7 +256,20 @@ object V2ScanRelationPushDown extends Rule[LogicalPlan] with PredicateHelper {
             sHolder.pushedLimit = Some(limitValue)
           }
           globalLimit
-        case _ => globalLimit
+        case _ =>
+          child transform {
+            case sort @ Sort(order, _, ScanOperation(_, filter, sHolder: ScanBuilderHolder))
+              if filter.length == 0 =>
+              val orders = DataSourceStrategy.translateSortOrders(order)
```

Comment:
1. The improve is pushDown in V2 Relation, Is it necessary to in the DataSourceV2Strategy
2. Optimizer rely the DataSourceStrategy, As if the upper layer depended on lower layer